crates/rolldown/src/utils/normalize_options.rs

```diff
@@ -83,6 +89,24 @@ pub fn normalize_options(mut raw_options: crate::BundlerOptions) -> NormalizeOpt
     experimental.strict_execution_order = Some(true);
   }
 
+  let inline_dynamic_imports = match raw_options.format {
```

Comment:
Normalize raw format first and match normalized format. No need to dealing with raw format here.